internal/ingest/upload.go

```diff
@@ -44,14 +46,24 @@ func (w *goaWrapper) UploadSip(
 		return nil, goaingest.MakeInvalidMultipartRequest(errors.New("invalid multipart request"))
 	}
 
+	// Identify file format to add extension in the object key.
+	// TODO: Use github.com/mholt/archives. We still use the archived github.com/mholt/archiver/v4
+	// in some activities, and using both causes a panic registering the same compressors.
+	format, stream, err := archiver.Identify(part.FileName(), part)
+	if err != nil {
+		return nil, goaingest.MakeInvalidMultipartRequest(errors.New("unable to identify format"))
+	}
```

Comment:
This is actually the same check we have in the extraction activity so, apart from giving us the extension it prevents the error in workflow. It also tries to identify the format from the stream alongside the filename, and detects `tar.xz` and other archive+compression formats.